addons/mail/controllers/thread.py

```diff
@@ -71,6 +76,10 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None):
         thread = request.env[thread_model]._get_from_request_or_raise(request, int(thread_id))
         if "body" in post_data:
             post_data["body"] = Markup(post_data["body"])  # contains HTML such as @mentions
+        new_partners = []
+        if 'partner_emails' in post_data:
+            new_partners = [record.id for record in request.env['res.partner']._find_or_create_from_emails(post_data['partner_emails'])]
```

Comment:
Just to make sure but it would be nice to add a test to ensure guest or portal users can't create arbitrary partners (it should crash for them as this is not done as sudo).

Reply:
Done!